src/main/java/duke/DataManager.java

```diff
@@ -0,0 +1,156 @@
+            switch(taskType) {
+            case TODO:
+                processTodoTask(task, tasks);
+                break;
+            case DEADLINE:
+                processDeadlineTask(task, tasks);
+                break;
+            case EVENT:
+                processEventTask(task, tasks);
+                break;
+            default:
+                return;
+            }
+        }
+    }
+
+    // If folder (data) and file (duke.txt) for saving and loading does not exist yet, create one for user.
+    private static void createSaveLocation() throws IOException {
+        // Create 'data' folder if it does not already exists
+        if (!Files.exists(DATA_DIRECTORY)) {
+            Files.createDirectory(DATA_DIRECTORY);
+        }
+        // Create 'duke.txt' file if it does not already exists
+        if (!Files.exists(DATA_PATH)) {
+            Files.createFile(DATA_PATH);
+        }
+    }
+
+    // Identifies the given task type
+    private static TaskType getTaskType(char taskTypeIndicator) throws DukeException {
+        // No breaks required as return statements are used
+        switch(taskTypeIndicator) {
+        case 'T':
+            return TaskType.TODO;
+        case 'D':
+            return TaskType.DEADLINE;
+        case 'E':
+            return TaskType.EVENT;
+        default:
+            throw new DukeException(ErrorTypeManager.ERROR_UNKNOWN_TASK_INDICATOR);
+        }
+    }
+
+    // Identify if task was marked as done before and mark it as done in current session if yes
+    private static void markTaskDoneIfDone (String task, ArrayList<Task> tasks) {
+        if (task.contains(Task.TICK_SYMBOL)) {
+            tasks.get(Task.getTaskCount() - 1).markAsDone();
+        }
+    }
+
+    // Exceptions are already handled in Duke.java and hence we will assume these are appropriate tasks
+    // No need to notify user that the task has been added too
+    private static void processTodoTask(String task, ArrayList<Task> tasks) {
+        String taskDescription = task.substring(START_OF_TASK_DESCRIPTION);
+
+        // Load saved to_do task into current tasks
+        tasks.add(new Todo(taskDescription));
+
+        // Recover its isDone status
+        markTaskDoneIfDone(task, tasks);
+    }
+
+    // Deadline tasks should be in the right format since exceptions were handled in Duke.java
+    // No need to notify user that the task has been added too
+    private static void processDeadlineTask(String task, ArrayList<Task> tasks) {
+        String taskDescription = task.substring(START_OF_TASK_DESCRIPTION);
+
+        String[] deadlineTaskParts = taskDescription.split(DEADLINE_TASK_DELIMITER);
+        String deadlineTaskDescription = deadlineTaskParts[0];
+        // Took the substring of the deadline portion because we want to ignore the closing bracket at the end
+        String deadline = deadlineTaskParts[1].substring(0, deadlineTaskParts[1].length() - 1);
```

Comment:
Since the magic number "1" is used in multiple places,  assigning it to a variable to explain what it does would value add to the developer and new devs by making its intention clear